.github/scripts/test-readme-check.sh

```diff
@@ -0,0 +1,128 @@
+fi
+
+# Colors for output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+print_success() {
+    echo -e "${GREEN}✅ $1${NC}"
+}
+
+print_error() {
+    echo -e "${RED}❌ $1${NC}"
+}
+
+print_warning() {
+    echo -e "${YELLOW}⚠️  $1${NC}"
+}
+
+print_header() {
+    echo "=================================================="
+    echo "$1"
+    echo "=================================================="
+}
+
+# Check if target directory is provided
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 [--ci] <component_dir|pipeline_dir>"
+    echo ""
+    echo "Options:"
+    echo "  --ci    Run in CI mode (non-interactive)"
+    echo ""
+    echo "Examples:"
+    echo "  $0 components/dev/hello_world"
+    echo "  $0 pipelines/training/my_pipeline"
+    echo "  $0 third_party/components/dev/external_component"
+    echo "  $0 --ci components/dev/hello_world  # CI mode"
+    exit 1
+fi
+
+TARGET_DIR=$1
+
+# Determine if it's a component or pipeline
+if [[ "$TARGET_DIR" == components/* ]] || [[ "$TARGET_DIR" == third_party/components/* ]]; then
+    TYPE_FLAG="--component"
+elif [[ "$TARGET_DIR" == pipelines/* ]] || [[ "$TARGET_DIR" == third_party/pipelines/* ]]; then
+    TYPE_FLAG="--pipeline"
+else
+    print_error "Invalid directory. Must be in components/, pipelines/, third_party/components/, or third_party/pipelines/"
+    exit 1
+fi
+
+print_header "Testing README validation for: $TARGET_DIR"
+
+# Backup existing README
+if [ -f "$TARGET_DIR/README.md" ]; then
+    print_warning "Backing up existing README..."
+    cp "$TARGET_DIR/README.md" "$TARGET_DIR/README.md.backup"
+else
+    print_warning "No existing README found"
+fi
+
+# Generate new README
+echo "Generating README..."
+uv run python -m scripts.generate_readme $TYPE_FLAG "$TARGET_DIR" --overwrite
+
+# Compare READMEs (ignore custom content section)
+if [ -f "$TARGET_DIR/README.md.backup" ]; then
+    echo "Comparing generated README with existing version..."
+    
+    # Extract content before custom-content marker from both files
+    awk '/<!-- custom-content -->/{exit} 1' "$TARGET_DIR/README.md.backup" > /tmp/old_readme.txt
+    awk '/<!-- custom-content -->/{exit} 1' "$TARGET_DIR/README.md" > /tmp/new_readme.txt
+    
+    if ! diff -u /tmp/old_readme.txt /tmp/new_readme.txt; then
+        print_error "README is out of sync for $TARGET_DIR"
+        echo ""
+        echo "The generated README differs from the committed version."
+        echo "Command to fix: uv run python -m scripts.generate_readme $TYPE_FLAG $TARGET_DIR --overwrite"
+        echo ""
+        echo "Diff saved to: /tmp/readme_diff.txt"
+        diff -u /tmp/old_readme.txt /tmp/new_readme.txt > /tmp/readme_diff.txt || true
+        
+        # In CI mode, always restore original README; otherwise ask user
+        if [ "$CI_MODE" = true ]; then
+            mv "$TARGET_DIR/README.md.backup" "$TARGET_DIR/README.md"
+            print_warning "Restored original README (CI mode)"
```

Comment:
@gmfrasca From the KEP, my understanding is that we should restore the original README and fail the job, prompting contributors to update or regenerate the README file. Could you update to fail the job in this case and update the message.